### ISSUE TITLE
Union key does not exist in object type

### ISSUE THREAD

**Issue by sergeysova** [JSMonk/hegel#65]
```ts

type Key = "demo" | "another" | "foo" | "bar"

const Foo = {
    another: 1,
    bar: 2,
    foo: 3,
    demo: 4,
}

function get(key: Key) {
    return Foo[key] // Error here
}
```

```
Property "  'another'
| 'bar'
| 'demo'
| 'foo'" does not exist in "{
	another: number,
	bar: number,
	demo: number,
	foo: number
}"
```

[Playground](https://jsmonk.github.io/hegel/try#FAFwngDgpgBA0lMMC8MBEATKBbA9mmAH3QEMA7XEACygCcDi0AzXfI9AIxPuGAGNcZAM4gYAMVYoYAb2Ax5McpRq0AXDACMAGjkKuamACYdCmC1zqAzCYVY86gCw6AvryYBXMnxABLQTABzKBAACgBrRHUEMABKGV15WmD3WjJxVgBtCLAAXWBXYCA)

**Comment by JSMonk** [comment]
Were fixed in 5baa37e08db2c61e154001ce09bfe4087032f2dc

**Comment by JSMonk** [comment]
Thank you for contribution ^_^
